### ISSUE TITLE
[Question]: Prepare test data for text classification

### ISSUE THREAD

**Issue by NDTanja** [flairNLP/flair#3113]
### Question

Hi,
I would like to train a TextClassifier and have created tab separated csv-files with the columns "text" and "label". I am using the CSVClassificationCorpus when building the corpus for the training. After training I get the message "ACHTUNG! No gold labels and no all_predicted_values found! Could be an error in your corpus or how you initialize the trainer!" Now I wonder how the test data must be prepared - unfortunately I find nothing in the documentation about it.

**Comment by helpmefindaname** [comment]
Hi @NDTanja 
can you give an example csv (e.g. one with only 1 or 2 rows) and the code you used to get to that warning?
I agree that the format is not documented well, but also it is very dynamic. If you post an example, I can help you specifically for your case.

**Comment by NDTanja** [comment]
Hi @helpmefindaname 
sorry for my late reply. Thank you for your help! 

**This is my code**:

from flair.data import Corpus
from flair.datasets import CSVClassificationCorpus
from flair.embeddings import WordEmbeddings, StackedEmbeddings, FlairEmbeddings, TransformerWordEmbeddings, \
    DocumentLSTMEmbeddings
from flair.models import TextClassifier
from flair.trainers import ModelTrainer

data_folder = r"./corpus"
column_name_map = {0: "text", 1: "label"}

corpus: Corpus = CSVClassificationCorpus(data_folder,
                                         column_name_map,
                                         skip_header=True,
                                         delimiter='\t',
                                         label_type='label')

label_type = 'example'
label_dict = corpus.make_label_dictionary(label_type='label')
print(label_dict)

sentence = corpus.test[0]
print(sentence)
sentence = corpus.test[1]
print(sentence)
sentence = corpus.test[2]
print(sentence)
sentence = corpus.test[3]
print(sentence)

embedding_types = [
    WordEmbeddings('de-crawl'),
    TransformerWordEmbeddings('bert-base-german-cased')
]
embeddings = StackedEmbeddings(embeddings=embedding_types)
word_embeddings = [WordEmbeddings('glove'), FlairEmbeddings('news-forward-fast'), FlairEmbeddings('news-backward-fast')]
document_embeddings = DocumentLSTMEmbeddings(word_embeddings, hidden_size=512, reproject_words=True,
                                             reproject_words_dimension=256)

clf = TextClassifier(document_embeddings, label_dictionary=label_dict, multi_label=False, label_type=label_type)

trainer = ModelTrainer(clf, corpus)

trainer.train('./results',
              learning_rate=0.1,
              mini_batch_size=32,
              max_epochs=2)

**This is the test-file**:
text	labels
abc	OR
def	OR
ghi 	OR
jkl	O

(these are only fake data)

**This ist the result test-file**:
abc
 - Gold: 
 - Pred: 

def
 - Gold: 
 - Pred: OR
 -> MISMATCH!

ghi
 - Gold: 
 - Pred: 

jkl
 - Gold: 
 - Pred:

**Comment by helpmefindaname** [comment]
hi @NDTanja this code loads the labels as the label type `label` but then you train the text classifier with the label type `example`.
As those mismatch, the labels that were loaded won't be used in training or evaluation.
if you allign the values, it should work

**Comment by NDTanja** [comment]
Hi @helpmefindaname oh... what a careless mistake. Thank you for your help, it works now!
